### PR TITLE
Finally Open Shell in SCF :D

### PR DESCRIPTION
**Pull request by charlyqchm**
Open Shell has been unified with SCF:
- Open Shell now is integrated in SCF.
- Modified dimensions of arrays in conver_subs to implement
  Open Shell.
- DFTB has been adapted for Open Shell.
- Solved problems with DIIS to converge DFTB in SCF.
- Added subroutines in dftb_subs to get some order in SCF.
- ECP and Open Shell are working together, but for some systems
  Open Shell explodes. This bug is not solve yet.

### CODE REVIEW

**Comment by charlyqchm** [comment]
I've been testing 2_FeH2O with CPU, and the convergence achieved with diis (default option for convergence), gives ridiculous results. I tested the same test with  hybrid_converg=t and sometimes I have good results, and sometimes it explodes or gives ridiculous results. Is this a problem with SCF? or is a problem with how LIO makes the calculations?

**Comment by fedepedron** [comment]
I believe I already tried this with current RC and it works either way, so it is a problem with your SCF I believe.

**Comment by ramirezfranciscof** [comment]
@fedepedron Your editing skills are faster than my mocking ones.

**Comment by fedepedron** [comment]
@ramirezfranciscof I'm the fastest regretter in the west.

File: lioamber/drive.f90 [hunk 1]
```diff
@@ -1030,13 +1040,15 @@ SUBROUTINE drive(ng2,ngDyn,ngdDyn)
...
+      if(OPEN.and.(.not.VCINP)) then
         allocate(rhoalpha(M*(M+1)/2),rhobeta(M*(M+1)/2))
-      else
+        rhoalpha = 0.0d0
+        rhobeta  = 0.0d0
+      else if (.not.OPEN) then
         allocate(rhoalpha(1),rhobeta(1))
+        rhoalpha = 0.0d0
+        rhobeta  = 0.0d0
       endif
```
[the remark this answers is not in the document]
Comment (fedepedron) [reply lioamber/drive.f90]: In fact, this broke any kind of closed shell restart XD

**Comment by fedepedron** [comment]
Fixed that for you, will run some tests on monday and if it works, it's in.

**Pull request merged**